controlplane/kubeadm/internal/workload_cluster_conditions.go

```diff
@@ -167,20 +244,45 @@ func (w *Workload) updateManagedEtcdConditions(ctx context.Context, controlPlane
 		unknownReason:     controlplanev1.EtcdClusterUnknownReason,
 		note:              "etcd member",
 	})
+
+	aggregateV1Beta2ConditionsFromMachinesToKCP(aggregateV1Beta2ConditionsFromMachinesToKCPInput{
+		controlPlane:      controlPlane,
+		machineConditions: []string{controlplanev1.KubeadmControlPlaneMachineEtcdMemberHealthyV1Beta2Condition},
+		kcpErrors:         kcpErrors,
+		condition:         controlplanev1.KubeadmControlPlaneEtcdClusterHealthyV1Beta2Condition,
+		falseReason:       controlplanev1.KubeadmControlPlaneEtcdClusterNotHealthyV1Beta2Reason,
+		unknownReason:     controlplanev1.KubeadmControlPlaneEtcdClusterHealthUnknownV1Beta2Reason,
+		trueReason:        controlplanev1.KubeadmControlPlaneEtcdClusterHealthyV1Beta2Reason,
+		note:              "etcd member",
+	})
 }
 
 func (w *Workload) getCurrentEtcdMembers(ctx context.Context, machine *clusterv1.Machine, nodeName string) ([]*etcd.Member, error) {
 	// Create the etcd Client for the etcd Pod scheduled on the Node
 	etcdClient, err := w.etcdClientGenerator.forFirstAvailableNode(ctx, []string{nodeName})
 	if err != nil {
 		conditions.MarkUnknown(machine, controlplanev1.MachineEtcdMemberHealthyCondition, controlplanev1.EtcdMemberInspectionFailedReason, "Failed to connect to the etcd pod on the %s node: %s", nodeName, err)
+
+		v1beta2conditions.Set(machine, metav1.Condition{
+			Type:    controlplanev1.KubeadmControlPlaneMachineEtcdMemberHealthyV1Beta2Condition,
+			Status:  metav1.ConditionUnknown,
+			Reason:  controlplanev1.KubeadmControlPlaneMachineEtcdMemberInspectionFailedV1Beta2Reason,
+			Message: fmt.Sprintf("Failed to connect to the etcd pod on the %s node: %s", nodeName, err),
```

Comment:
Just wondering if we really want to put this error in the message. But fine for me if you're sure

Seems like we do this everywhere here. I guess then it's fine? (as we didn't hear about issues with it before?)

Reply:
I'm not aware of issues.
AFAIK sometimes those errors can give a pointer about the underlying connectivity problem, so I think we should keep for now